bakery/mgostorage/rootkey.go

```diff
@@ -0,0 +1,344 @@
+package mgostorage
+
+import (
+	"crypto/rand"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/juju/loggo"
+	"gopkg.in/errgo.v1"
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+
+	"gopkg.in/macaroon-bakery.v1/bakery"
+)
+
+// Functions defined as variables so they can be overidden
+// for testing.
+var (
+	timeNow = time.Now
+
+	mgoCollectionFindId = (*mgo.Collection).FindId
+)
+
+var logger = loggo.GetLogger("bakery.mgostorage")
+
+// maxPolicyCache holds the maximum number
+// of storage policies that can hold cached keys in
+// a given RootKeys instance.
+// In practice, this could probably be 3 or 4 and it
```

Comment:
I'm not sure this line tells me anything. Should I change this number to 3 or 4?